src/py_converters.cpp

```diff
@@ -130,7 +130,17 @@ int convert_cap(PyObject *capobj, void *capp)
     int values[] = {agg::butt_cap, agg::round_cap, agg::square_cap};
     int result = agg::butt_cap;
 
-    if (!convert_string_enum(capobj, "capstyle", names, values, &result)) {
+    PyObject *name = PyUnicode_FromString("name");
+    PyObject *cap_string = PyObject_GetAttr(capobj, name);
```

Comment:
you can probably use https://docs.python.org/3/c-api/object.html#c.PyObject_GetAttrString instead.

Reply:
Will fix in the future, for now, Tom has asked that I revert a large chunk of this to make it so that we don't expose the JoinStyle and CapStyle classes publicly *at all* for now.